### DISCUSSION TITLE
Use python-socks instead of socksio

### DISCUSSION THREAD

**Discussion by byehack** [encode/httpcore#667] [Ideas]
benefits:
1. active maintainer
2. custom async backends (curio, trio, anyio, asyncio)
3. socks5h
4. https proxies
5. httpx extension available (when implemented in httpcore maybe no more need this.)

**Comment by lovelydinosaur** [comment]
what issue do you want to resolve?

start with one and we'll talk it through...

**Reply by dalf** [reply]
Would python-socks help to implement HTTPS proxy support (TLS-in-TLS) ?

https://github.com/encode/httpx/issues/1434#issuecomment-979193104

**Reply by byehack** [reply]
I switched to `httpx-socks` last year, I don't remember correctly, but there were few issues that I forgot report them.

**Reply by byehack** [reply]
it seems httpx not allowed for socks4:
https://github.com/encode/httpx/blob/579a3f2fb8e1ebd151b8f3a6c1e9bc6dedf5f5bd/httpx/_config.py#L335-L336
vs 
https://github.com/romis2012/python-socks/blob/b834fa499b383dc81daf573031eb859b7de34869/python_socks/_helpers.py#L49-L55
